app/models/miq_report/seeding.rb

```diff
@@ -47,17 +47,28 @@ def seed_record(path, report)
         _log.info("#{report.new_record? ? "Creating" : "Updating"} MiqReport #{filename.inspect}")
 
         yml   = YAML.load_file(path).symbolize_keys
+        name = yml[:menu_name].strip
+
         attrs = yml.slice(*column_names_symbols)
         attrs.delete(:id)
         attrs[:filename]      = filename
         attrs[:file_mtime]    = mtime
-        attrs[:name]          = yml[:menu_name].strip
+        attrs[:name]          = name
         attrs[:priority]      = File.basename(path).split("_").first.to_i
         attrs[:rpt_group]     = File.basename(File.dirname(path)).split("_").last
         attrs[:rpt_type]      = "Default"
         attrs[:template_type] = path.start_with?(REPORT_DIR.to_s) ? "report" : "compare"
 
+        begin
         report.update_attributes!(attrs)
+        rescue ActiveRecord::RecordInvalid
+          duplicate = find_by(:name => name)
```

Comment:
Thankfully this should be only happening on a rare occasion, so shouldn't be a big deal as far as adding to the N+1 count.  👍 

That said, if we wanted, we could maybe raise these all at the end, and do a single find query.  Probably not worth it, just a thought.